redbot/cogs/mod/kickban.py

```diff
@@ -174,6 +174,30 @@ class KickBanMixin(MixinMeta):
             return
         audit_reason = get_audit_reason(author, reason)
         try:
```

Comment:
This try shouldn't really be here if you're sending the message before the kick, since there might be a situation where the message will send successfully, and then the kick itself will fail.
Since there already is a role hierarchy check on lines 163-171, and proper decorators for bot and user permissions are used, the `except` case with `discord.Forbidden` should be removed since it will never have a chance to run after the changes (if I'm wrong on this one, please correct me - hierarchy and permissions should be the only cause of Forbidden here, and these are covered).

In addition, the `try` statement should be moved to right before the kick happens, to make sure it's guarding the right / intended piece of code.

Reply:
Unless I'm getting it wrong, kick can't fail because the command has a `@commands.bot_has_permissions(kick_members=True)` check.